File: 532-k-diff-pairs-in-an-array/k-diff-pairs-in-an-array.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        c = Counter(nums)
        count = 0

        if (k == 0):
            for key, val in c.items():
                if val > 1:
                    count += 1
        else:
            for key, val in c.items():
                if (key + k in c):
                    count += 1

        return count


        

File: 532-k-diff-pairs-in-an-array/k-diff-pairs-in-an-array.js

var findPairs = function(nums, k) {
    let lookup = {}
    
    for (let num of nums) {
        if (!lookup[num]) lookup[num] = 0
        lookup[num] += 1
    }

    let count = 0
    
    for ([key, val] of Object.entries(lookup)) {
        }

    if (k == 0) {
        for ([key, val] of Object.entries(lookup)) {
            if (val > 1) count +=1
        }
        
    }
    else {
        for ([key, val] of Object.entries(lookup)) {
            if (lookup[Number(key) + k]) count ++
        }
    }
    return count
};